airbyte-webapp/src/components/ui/NextTable/NextTable.tsx

```diff
@@ -1,43 +1,72 @@
-import { ColumnDef, flexRender, useReactTable, getCoreRowModel } from "@tanstack/react-table";
+import { ColumnDef, flexRender, useReactTable, getCoreRowModel, ColumnSort } from "@tanstack/react-table";
 import classNames from "classnames";
 import { PropsWithChildren } from "react";
 
 import styles from "./NextTable.module.scss";
 
-export interface TableProps<TData> {
-  data: TData[];
-  columns: Array<ColumnDef<TData>>;
-  onClickRow?: (data: unknown) => void;
+export interface TableProps<T> {
   className?: string;
+  // We can leave type any here since useReactTable options.columns itself is waiting for Array<ColumnDef<T, any>>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  columns: Array<ColumnDef<T, any>>;
+  data: T[];
+  erroredRows?: boolean;
```

Comment:
I guess it came from an old design(probably) - For the case when the connection sync failed and we needed to color it red. And I also stand for removing it, since it doesn't make any effect.